packages/syft/src/syft/service/action/action_permissions.py

```diff
@@ -46,7 +46,10 @@ def permission_string(self) -> str:
         if self.permission in COMPOUND_ACTION_PERMISSION:
             return f"{self.permission.name}"
         else:
-            return f"{self.credentials.verify}_{self.permission.name}"
+            if self.credentials is not None:
+                return f"{self.credentials.verify}_{self.permission.name}"
+            else:
+                return f"{self.permission.name}"
```

Comment:
this can be unnested

```py
if self.credentials is not None:
    return f"{self.credentials.verify}_{self.permission.name}"
return f"{self.permission.name}"
```